### PR TITLE
fix: search go module when update and no -module provide

### LINKED ISSUE #40: Crash when type is HTTP and module not set
**Describe the bug**

Crash when type is HTTP and module not set

**To Reproduce**

```protobuf
syntax = "proto3";

package biz.hello;

// The greeting service definition.
option go_package = "biz/hello";

message Request {
  string message = 1;
}

message Response {
  string message = 1;
}

service Echo {
  rpc Echo (Request) returns (Response) {}
}
```

```shell
➜ cwgo server -service=Host -type HTTP -I idl -idl idl/hello/hello.proto
plugin protoc_gen_hertz returns error: exit status 1, cause:
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x10 pc=0x100adb498]

goroutine 1 [running]:
github.com/cloudwego/hertz/cmd/hz/protobuf.(*Plugin).Run.func1()
	/Users/rogers/go/pkg/mod/github.com/cloudwego/hertz/cmd/hz@v0.5.3-0.20230208061226-42701b832677/protobuf/plugin.go:86 +0x28
github.com/cloudwego/hertz/cmd/hz/protobuf.(*Plugin).Run(0x14000121eb8)
	/Users/rogers/go/pkg/mod/github.com/cloudwego/hertz/cmd/hz@v0.5.3-0.20230208061226-42701b832677/protobuf/plugin.go:115 +0x518
github.com/cloudwego/hertz/cmd/hz/app.PluginMode()
	/Users/rogers/go/pkg/mod/github.com/cloudwego/hertz/cmd/hz@v0.5.3-0.20230208061226-42701b832677/app/app.go:130 +0x9c
main.main()
	/Users/rogers/go/pkg/mod/github.com/cloudwego/cwgo@v0.0.0-20230224083820-c081b37bf35f/cwgo.go:32 +0x20
--hertz_out: protoc-gen-hertz: Plugin failed with status code 2.
```

**Expected behavior**

**Screenshots**

**Kitex version:**

Please provide the version of Kitex you are using.

**Environment:**

The output of `go env`.

**Additional context**

### PR DESCRIPTION
**Pull request by rogerogers**
#### What type of PR is this?

fix: search go module when update and no -module provide

#### What this PR does / why we need it (en: English/zh: Chinese):

en: search go module when update and no -module provide
zh: 更新的时候自动查找go.mod来赋值module参数

#### Which issue(s) this PR fixes:

Fixes #40

### CODE REVIEW

File: pkg/server/server.go [hunk 1]
```diff
@@ -99,6 +99,20 @@ func Server(c *config.ServerArgument) error {
...
+			module, path, ok := util.SearchGoMod(".", false)
+			if ok {
+				// go.mod exists
+				if c.GoMod != "" && module != c.GoMod {
+					return fmt.Errorf("module name given by the '-module' option ('%s') is not consist with the name defined in go.mod ('%s' from %s)", c.GoMod, module, path)
+				}
+				args.Gomod = module
+			} else {
+				workPath, err := filepath.Abs(".")
+				if err != nil {
```
Comment (FGYFFFF) [inline pkg/server/server.go:111]: why add "workPath", but not use?

Comment (rogerogers) [reply pkg/server/server.go:111]: > why add "workPath", but not use?

`return fmt.Errorf("go.mod not found in %s", workPath)` for print error logs.

Comment (FGYFFFF) [reply pkg/server/server.go:111]: 上面那个 path 应该就是 "workPath"。另外问下，这里需要向上层文件里去找 "go.mod" 吗？

Comment (rogerogers) [reply pkg/server/server.go:111]: 那个path还是相对路径，这里想着打印绝对路径提示下。

**Review by li-jin-gou — APPROVED** [review]

**Pull request merged**